### PR TITLE
fix: exchange transactions are modeled by Exchange.transfer pallet calls

### PR DESCRIPTION
**Pull request by georgiosd**
fixes ENGIN-1036

@garrettmaring please confirm this is correct. I also noticed the `sell(to, amount)` call but seems to be an internal call for other purposes.

### CODE REVIEW

File: engi-server/Indexing/TransactionIndex.cs [hunk 1]
```diff
@@ -106,7 +106,10 @@ public static bool GetIsSuccessful(string pallet, dynamic events)
 
             if(pallet == ""Exchange"")
             {
-                return TransactionType.Exchange;
+                if(call == ""transfer"")
```
Comment (garrettmaring) [inline engi-server/Indexing/TransactionIndex.cs:109]: i think we need to index "sell" extrinsics too. `sell` is the native tx users call to sell engi, `transfer` is called internally in response to the bridge seeing that engi was *bought* on ethereum

cc @tjsharp1

Comment (georgiosd) [reply engi-server/Indexing/TransactionIndex.cs:109]: Actually that makes sense, I already have logic in `GetAmount` for `sell` - sorry I didn't notice it. Submitting fix shortly.

**Comment by georgiosd** [comment]
Adjusted as per above, please review/approve @garrettmaring

**Review by garrettmaring — APPROVED** [review]

**Comment by garrettmaring** [comment]
we should have two exchange types (buy, sell):

* buy == transfer from orgin chain to buyer https://github.com/engi-network/blockchain/blob/main/pallets/exchange/src/lib.rs#L313
* sell is as you have it now

**Comment by georgiosd** [comment]
@garrettmaring it would be best to fix this with a couple of real transactions. Could you please provide two block numbers in staging with a transfer each way and the expected result from the index? This is pretty important so I don't wanna mess it up.

**Comment by georgiosd** [comment]
Closing in favor of new PR

**Pull request closed without merging**